src/node_url.h

```diff
@@ -183,6 +183,7 @@ class URL {
   URL() : URL("") {}
 
  private:
+  Environment* env_ = nullptr;
```

Comment:
This doesn’t seem quite right – why isn’t this passed along like other arguments? It doesn’t conceptually make a lot of sense to say that a URL is bound to a specific Environment instance, when it doesn’t associate with JS objects.